ui/index.d.ts

```diff
@@ -1188,6 +1203,35 @@ export interface UIExtensionVariables {
     ignoreBundle?: boolean;
 }
 
+/**
+ * Interface for experimentation service adapter
+ */
+export interface IExperimentationServiceAdapter {
+    /**
+     * Gets whether or not the flight is enabled from the cache (which will be ~1 session delayed)
+     * @param flight The flight variable name
+     */
+    isCachedFlightEnabled(flight: string): Promise<boolean>;
+
+    /**
+     * Gets whether or not the flight is enabled directly from the web. This is slower than cache and can result in behavior changing mid-session.
+     * @param flight The flight variable name
+     */
+    isLiveFlightEnabled(flight: string): Promise<boolean>;
+
+    /**
+     * Gets a treatment variable from the cache (which will be ~1 session delayed)
+     * @param name The variable name
+     */
+    getCachedTreatmentVariable<T extends string | number | boolean>(name: string): T | undefined;
```

Comment:
I would like for this to be `async` like `isCachedFlightEnabled` but I can't get TypeScript to let me. It's saying it can't assign `string` to `string | number | boolean | undefined`.

Reply:
Figured it out!